qp/utils.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import sys
+
+def safelog(arr, threshold=sys.float_info.epsilon):
+    """
+    Takes log of array with zeroes.
+
+    Parameters
+    ----------
+    arr: ndarray
+        Values to be logged
+    threshold: float
+        Small, positive value to replace zeros and negative numbers
+
+    Returns
+    -------
+    logged: ndarray
+        Logged values, with approximation in place of zeros and negative numbers
+    """
+    shape = np.shape(arr)
+    flat = arr.flatten()
+    logged = np.log(np.array([max(a,threshold) for a in flat])).reshape(shape)
+    return logged
+
+def calckl(qn, pn):
+    """
+    Calculates Kullback-Leibler Divergence
+
+    Parameters
+    ----------
+    qn: ndarray
```

Comment:
Okay, that makes sense.  Thanks for clarifying!  I created #23 so the PDF object has some histogram representation option and assigned the issue to myself.  I think I should probably make that change after this pull request is resolved because it's big enough to have problems of its own.  Is a KLD that's hardcoded to be relative to the truth sufficient for now?